runtime/dart_isolate.cc

```diff
@@ -678,7 +686,13 @@ fxl::RefPtr<DartSnapshot> DartIsolate::GetIsolateSnapshot() const {
 }
 
 fml::WeakPtr<DartIsolate> DartIsolate::GetWeakIsolatePtr() const {
-  return weak_factory_.GetWeakPtr();
+  return weak_factory_ ? weak_factory_->GetWeakPtr()
+                       : fml::WeakPtr<DartIsolate>();
+}
+
+void DartIsolate::ResetWeakPtrFactory() {
+  FXL_CHECK(weak_factory_);
```

Comment:
Oh, I meant RestWeakPtrFactoryIfExists instead of adding an assertion. Future callers of this method should not have to care if a factory has already been reset.

Reply:
ResetWeakPtrFactory will clear the unique_ptr<WeakPtrFactory>, destructing the WeakPtrFactory if one existed

Reply:
Sorry. Just noticed it is a unique ptr. My bad. The simple reset is good.